porcupine/menubar.py

```diff
@@ -328,6 +330,31 @@ def setup() -> None:
     set_enabled_based_on_tab(path, (lambda tab: isinstance(tab, tabs.FileTab)))
 
 
+def add_filetab_action(path: str, action: actions.FileTabAction, **kwargs: Any) -> None:
+    """
+    This is a convenience function that does several things:
+
+    * Create a menu item at the given path.
+    * Ensure the menu item is enabled only when the selected tab is a
+      :class:`~porcupine.tabs.FileTab` AND when
+      :class:`~porcupine.actions.FileTabAction.availability_callback`
+      evaluates to True.
+    * Run :class:`~porcupine.actions.FileTabAction.callback` when the
+      menu item is clicked.
+
+    The ``callback`` is called with the selected tab as the only
+    argument when the menu item is clicked.
+
+    You usually don't need to provide any keyword arguments in ``**kwargs``,
+    but if you do, they are passed to :meth:`tkinter.Menu.add_command`.
+    """
+    menu_path, item_text = _split_parent(path)
```

Comment:
Presumably we could take only the `menu_path` as an argument and infer `item_text` through the action object.

Reply:
Maybe I should clarify that `_split_parent()` does `"Foo/Bar/Baz"` --> `("Foo/Bar", "Baz")` conversion. Doctests would be very useful here :)

Reply:
so lets talk about that.

I assume you mean use the `Action.name` attribute. That does make sense at first blush, my concern is does that limit us to what we can put in the menu item, things like formatting or something. I think it's probably fine though?

Reply:
Other problem I see is that this changes the API, but actually it doesn't have to. If you give a "fully qualified" path, like `topmenu/submenu/itemname` then we treat it like we always have, but if you just do `topmenu/submenu/` with the trailing `/`, then we use the name.

Reply:
> limit us to what we can put in the menu item

My plan for the menubar is:
- Use this function for almost everything
- Expose the menus as-is for custom stuff (e.g. funny named items, items that are more like checkboxes like the filetype chooser, anything else custom)

So if this function covers 99% of use cases, that's enough :)

Reply:
I didn't read your message till I already did it, it would be easy enough to undo though: 0e77816

Reply:
And for whatever reason, I didn't tell you my thoughts about trailing `/` magic :)

I think people expect a trailing `/` to do nothing, like how `ls /home/akuli/` is really `ls /home/akuli`. While a little bit of trailing slash magic is fine, e.g. `ls file.txt/` does not work, this goes a bit too far IMO.

Reply:
Okay, so to be clear, the only supported way to define the label text for `add_filetab_action` will be via `Action.name`?

Reply:
I think that would be good, yes.